File: ML/visheshks04_VisheshKumarSingh_2024csit1105_2/GettingStarted/GCDAndLCM.cpp
#include <iostream>
using namespace std;

int gcd(int, int);
int lcm(int, int);

int main(int argc, char **argv){
    int num1, num2;
    cin >> num1 >> num2;

    cout<<gcd(num1, num2)<<endl;
    cout<<lcm(num1, num2)<<endl;
    
    return 0;
}

int gcd(int a, int b){
    int min = a>b?b:a;
    
    for(int i=min;i>1;i--){
        if(a%i == 0 && b%i == 0)
            return i;
    }
    
    return 1;
}

int lcm(int a, int b){
    int max = a>b?a:b;
    int min = a>b?b:a;
    
    for(int i=1;i<=a*b;i++){
        if((max*i)%min == 0)
            return max*i;
    }
    
    return a*b;
}

File: ML/ambuj-1211_AmbujKulshreshtha_2024csit1160_2/GettingStarted/gcdandlcm.java
import java.util.*;
public class gcdandlcm {
    public static void main(String[] args){
    int num1,num2,gcd=0,lcm=0;
    Scanner sc=new Scanner(System.in);
    num1=sc.nextInt();
    num2=sc.nextInt();
    if(num1<num2){
    for(int i=1;i<=num1/2;i++){
        if(num1%i==0&&num2%i==0)
        gcd=i;
    }
    }else
    if(num2<num1){
        for(int i=1;i<=num2/2;i++){
            if(num1%i==0&&num2%i==0)
             gcd=i;
        }
    }
    lcm=(num1*num2)/gcd;
    System.out.println(gcd);
    System.out.println(lcm);
    sc.close();
}}
